Slim/Player/ProtocolHandlers.pm

```diff
@@ -101,8 +101,17 @@ sub handlerForURL {
 	}
 
 	# Load the handler when requested..
-	my $handler = $class->loadHandler($protocol);
-	
+	my $handler;
+	foreach my $key ( keys %protocolHandlers ) {
+		next unless $key =~ m/^\(/;
```

Comment:
This check would require the regex to do grouping (`/(...)/`) why so?

Reply:
I started off with `next unless ref($key) eq 'Regexp'`, but it didn’t work.  It turns out that it _seems_ like you can use compiled regular expressions as hash keys, but actually you can’t.  What happens is that the regular expression gets cast to a string, and the string-form of the regular expression is always wrapped with `(? … )`:

```console
> perl -MData::Dumper -E '$key = qr/foo/; $hash{$key} = 1; say Dumper $key, keys(%hash)'
$VAR1 = qr/foo/u;
$VAR2 = '(?^u:foo)';
```

I wanted to get started with the simplest thing that would allow me to work on a plugin and prove that this idea works.  But, for performance, we might decide to store regular expression handlers in a new structure, `@urlHandlers` say, rather than the existing `%protocolHandlers`.  I don’t think that this would need to affect the API we offer to plugins.

Reply:
I thought about another list, too. And then you could use [`use Tie::RegexpHash;`](https://metacpan.org/pod/Tie::RegexpHash) to use those regexes more naturally (see eg. `Slim::Formats::RemoteMetadata`).